test/e2e/canary_test.go

```diff
@@ -156,12 +208,12 @@ spec:
 		ScaleRollout(8).
 		WaitForRolloutReplicas(10).
 		Then().
-		// NOTE: the numbers below may change in the future.
-		// See: https://github.com/argoproj/argo-rollouts/issues/738
 		ExpectCanaryStablePodCount(6, 4).
 		When().
-		ScaleRollout(4)
-	// WaitForRolloutReplicas(4) // this doesn't work yet (bug)
+		ScaleRollout(4).
+	    WaitForRolloutReplicas(8).
+		Then().
+		ExpectCanaryStablePodCount(4, 4)
```

Comment:
I don't understand the fix. Why after scaling down the spec.replicas, do we still have 8 pods?

Reply:
fixed the issue. Thanks for the comment